### PR TITLE
xalan-c 1.11: New formula

### PR DESCRIPTION
**Pull request by rleigh-dundee**
The Xalan C++ XSLT processor based on Xerces-C++.  This uses the already packaged xerces-c formula.

### CODE REVIEW

File: Library/Formula/xalan-c.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      Authors: <xsl:apply-templates select="/Article/Authors/Author"/>
+      </xsl:template>
+        <xsl:template match="Author">
+      * <xsl:value-of select="." />
+        </xsl:template>
+      </xsl:stylesheet>
+    EOS
+
+    assert_match "Article: An XSLT test-case\nAuthors: \n* Roger Leigh\n* Open Microscopy Environment", shell_output("#{bin}/Xalan #{testpath}/input.xml #{testpath}/transform.xsl")
+
```
Comment (bfontaine) [inline Library/Formula/xalan-c.rb]: Please remove this empty line.

File: Library/Formula/xalan-c.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    ENV["XALAN_LOCALE_SYSTEM"] = "inmem"
+    ENV["XALAN_LOCALE"] = "en_US"
+
+    Dir.chdir("c") do
+      system "./configure", "--disable-dependency-tracking",
+                            "--disable-silent-rules",
+                            "--prefix=#{prefix}"
+      system "make", "install"
+      # Clean up links
+      Dir.glob("#{lib}/*.dylib.*") { |p| rm p }
```
Comment (bfontaine) [inline Library/Formula/xalan-c.rb]: use `rm Dir["#{lib}/*.dylib.*"]`.

File: Library/Formula/xalan-c.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  sha256 "4f5e7f75733d72e30a2165f9fdb9371831cf6ff0d1997b1fb64cdd5dc2126a28"
+
+  option :universal
+
+  depends_on "xerces-c"
+
+  def install
+    ENV.deparallelize # This build isn't parallel safe.
+    ENV.universal_binary if build.universal?
+    ENV["XALANCROOT"] = "#{Dir.pwd}/c"
```
Comment (bfontaine) [inline Library/Formula/xalan-c.rb]: You can use `buildpath` instead of `Dir.pwd`.

File: Library/Formula/xalan-c.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  homepage "https://xalan.apache.org/xalan-c/"
+  url "https://www.apache.org/dyn/closer.cgi?path=xalan/xalan-c/sources/xalan_c-1.11-src.tar.gz"
+  sha256 "4f5e7f75733d72e30a2165f9fdb9371831cf6ff0d1997b1fb64cdd5dc2126a28"
+
+  option :universal
+
+  depends_on "xerces-c"
+
+  def install
+    ENV.deparallelize # This build isn't parallel safe.
```
Comment (bfontaine) [inline Library/Formula/xalan-c.rb]: Is upstream aware of that?

Comment (rleigh-dundee) [inline Library/Formula/xalan-c.rb]: I'll check and file a bug if they are not.

Comment (bfontaine) [inline Library/Formula/xalan-c.rb]: Thanks!

Comment (rleigh-dundee) [inline Library/Formula/xalan-c.rb]: https://issues.apache.org/jira/browse/XALANC-696

**Comment by rleigh-dundee** [comment]
@bfontaine Thanks for the suggestions; I've fixed all these now and pushed the result.

File: Library/Formula/xalan-c.rb (hunk 4 above)
Comment (dunn) [inline Library/Formula/xalan-c.rb]: Can you add that in a comment about `ENV.deparallelize`, so we know if/when to remove it? Thanks!

**Comment by rleigh-dundee** [comment]
I've added a one-line patch from Debian to fix a potential segfault.

File: Library/Formula/xalan-c.rb (hunk 4 above)
Comment (rleigh-dundee) [inline Library/Formula/xalan-c.rb]: @dunn No problem, I've made and pushed that change.

File: Library/Formula/xalan-c.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  homepage "https://xalan.apache.org/xalan-c/"
+  url "https://www.apache.org/dyn/closer.cgi?path=xalan/xalan-c/sources/xalan_c-1.11-src.tar.gz"
+  sha256 "4f5e7f75733d72e30a2165f9fdb9371831cf6ff0d1997b1fb64cdd5dc2126a28"
+
+  option :universal
+
+  depends_on "xerces-c"
+
+  # Fix segfault
+  patch :DATA
```
Comment (dunn) [inline Library/Formula/xalan-c.rb]: Is there an upstream ticket corresponding to the Debian patch?

Comment (rleigh-dundee) [inline Library/Formula/xalan-c.rb]: Yes.  I can add a link to that also.

Comment (rleigh-dundee) [inline Library/Formula/xalan-c.rb]: (now committed and pushed)

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for simple version updates is `xalan-c 1.11 (new formula)`.

**Pull request closed without merging**